### PR TITLE
address high-pt efficiency drop in EB EG objects

### PR DESCRIPTION
**Pull request by cecilecaillol**
#### PR description:

Fix for the high pt electron/photon efficiency in the barrel, which was due to hit saturation.
Now the shower shape and calo iso criteria are not applied for objects with a reco pT above 130 GeV.
See the presentation in https://indico.cern.ch/event/995146/

#### PR validation:

Efficiencies and rates checked in Zprime sample, no drop in efficiency anymore and negligible increase in rates.

@cerminar

### CODE REVIEW

File: L1Trigger/L1CaloTrigger/plugins/L1EGammaCrystalsEmulatorProducer.cc [hunk 1]
```diff
@@ -1176,28 +1176,45 @@ void L1EGCrystalClusterEmulatorProducer::produce(edm::Event& iEvent, const edm::
...
     if (!((a0_80 - a1_80 * pt) > iso))
-      return false;
+      is_iso = false;
   } else {
     if (iso > a0)
-      return false;
+      is_iso = false;
   }
-  return true;
+  if (pt > 130)
```
Comment (kpedro88) [inline L1Trigger/L1CaloTrigger/plugins/L1EGammaCrystalsEmulatorProducer.cc:1187]: please define a clearly named constant and use it in place of magic number 130 everywhere

**Comment by kpedro88** [comment]
please test

**Comment by srimanob** [comment]
Please test

**Comment by srimanob** [comment]
@cecilecaillol 
Do we have plots in DQM showing what you showed in the presentation?

**Comment by srimanob** [comment]
@cecilecaillol 
Any comment on my last question? Thanks.

@rekovic Could you please review for L1? Thanks.

**Comment by cecilecaillol** [comment]
The results of my checks are in https://github.com/cms-sw/cmssw/pull/32697. Other people have confirmed the efficiency drop is fixed with an independent framework.

**Comment by srimanob** [comment]
+Upgrade

Thanks.

**Comment by silviodonato** [comment]
@cecilecaillol I have just a minor comment. Shouldn't `plateau_ss` be a configurable parameter (ie. taken with iConfig.getParameter)?
@rekovic please review/sign

**Comment by cecilecaillol** [comment]
The plateau value is determined by the threshold at which the hits saturate, which will not change, so I dont think it is worth/justified making it a configurable parameter.

**Comment by silviodonato** [comment]
@rekovic please review/sign

**Comment by rekovic** [comment]
+1

**Comment by qliphy** [comment]
+1

**Pull request merged**